File: ui/Pause/Pause.cs
using Godot;

public class Pause : Node2D
{
    public Node2D Overlay()
    {
        return GetNode<Node2D>("Overlay");
    }

    public override void _EnterTree()
    {
        base._EnterTree();
        Overlay().Visible = false;
    }

    public override void _Ready()
    {
        base._Ready();
        Overlay().Visible = false;
    }

    public void OnPauseButtonPressed()
    {
        GetTree().SetInputAsHandled();
    }
}

File: ui/Pause/Pause.ts
export class Pause extends Node2D {
  get Overlay() {
    return this.get_node("Overlay");
  }

  _enter_tree() {
    this.Overlay.visible = false;
  }

  _ready() {
    this.Overlay.visible = false;
  }

  onPauseButtonPressed() {
    this.get_tree().set_input_as_handled();
  }
}
